### PR TITLE
feat: add in memory web provider

### PR DESCRIPTION
**Pull request by moredip**
## This PR
- adds a second version of the in-memory provider which supports the web SDK.
 
### Notes
I moved the original provider (built against the js-sdk) into a `node` folder, then created a parallel implementation in a `web` folder. The two providers are exported as `NodeInMemoryProvider` and `WebInMemoryProvider`.

### CODE REVIEW

File: libs/providers/in-memory/src/lib/web/in-memory-provider.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    const flagValue = this.lookupFlagValueOrThrow(flagKey);
+
+    if (typeof flagValue !== 'boolean') {
+      throw new TypeMismatchError();
+    }
+
+    return staticResolution(flagValue);
+  }
+
+  resolveStringEvaluation(flagKey: string, _defaultValue: string): ResolutionDetails<string> {
```
Comment (beeme1mr) [inline libs/providers/in-memory/src/lib/web/in-memory-provider.ts:35]:
```suggestion
  resolveStringEvaluation(flagKey: string, _: string): ResolutionDetails<string> {
```

File: libs/providers/in-memory/src/lib/web/in-memory-provider.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    const flagValue = this.lookupFlagValueOrThrow(flagKey);
+
+    if (typeof flagValue !== 'number') {
+      throw new TypeMismatchError();
+    }
+
+    return staticResolution(flagValue);
+  }
+
+  resolveObjectEvaluation<T extends JsonValue>(flagKey: string, defaultValue: T): ResolutionDetails<T> {
```
Comment (beeme1mr) [inline libs/providers/in-memory/src/lib/web/in-memory-provider.ts:55]:
```suggestion
  resolveObjectEvaluation<T extends JsonValue>(flagKey: string, _: T): ResolutionDetails<T> {
```

File: libs/providers/in-memory/src/lib/web/in-memory-provider.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  replaceConfiguration(flagConfiguration: FlagConfiguration) {
+    const flagsChanged = Object.entries(flagConfiguration)
+      .filter(([key, value]) => this._flagConfiguration[key] !== value)
+      .map(([key]) => key);
+
+    this._flagConfiguration = { ...flagConfiguration };
+    this.events.emit(ProviderEvents.ConfigurationChanged, { flagsChanged });
+  }
+
+  resolveBooleanEvaluation(flagKey: string, _defaultValue: boolean): ResolutionDetails<boolean> {
```
Comment (beeme1mr) [inline libs/providers/in-memory/src/lib/web/in-memory-provider.ts:25]:
```suggestion
  resolveBooleanEvaluation(flagKey: string, _: boolean): ResolutionDetails<boolean> {
```

**Comment by toddbaert** [comment]
I'm in favor of having this, but with the addition of [this](https://github.com/open-feature/spec/blob/main/specification/appendix-A.md), we can put this directly in the SDK now.

**Comment by beeme1mr** [comment]
We now have an in-memory provider included in the SDK.
https://github.com/open-feature/js-sdk/tree/main/packages/client/src/provider/in-memory-provider

**Pull request closed without merging**